pkg/cluster/internal/providers/common/provision.go

```diff
@@ -80,18 +79,10 @@ func RunContainer(engine, name string, args []string, opts ...RunContainerOpt) e
 		logCtx := context.Background()
 		logCtx, logCancel := context.WithTimeout(logCtx, 30*time.Second)
 		defer logCancel()
-		// use os/exec.CommandContext directly, as kind/pkg/exec.CommandContext lacks support for killing
```

Comment:
NOTE: ctx.Cancel() for a CommandContext already kills the process. We don't need to call killl directly, we should leverage cancelling the context instead (and we do).